File: src/main/java/org/shark/example/service/mongo/MongoCreateService.java
package org.shark.example.service.mongo;

import lombok.RequiredArgsConstructor;
import org.shark.example.datasource.mongo.ExampleRepository;
import org.shark.example.datasource.mongo.pojo.ExampleDo;
import org.shark.example.service.pojo.ResponseDto;
import org.springframework.stereotype.Service;

@RequiredArgsConstructor
@Service
public class MongoCreateService {

    private final ExampleRepository exampleRepository;

    public ResponseDto<Void> start(ExampleDo exampleDo) {
        exampleRepository.save(exampleDo);
        return ResponseDto.<Void>builder().status(true).build();
    }

}

File: src/main/kotlin/org/shark/example/service/mongo/MongoCreateService.kt
package org.shark.example.service.mongo

import org.shark.example.datasource.mongo.ExampleRepository
import org.shark.example.datasource.mongo.pojo.ExampleDo
import org.shark.example.service.pojo.ResponseDto
import org.springframework.stereotype.Service

@Service
class MongoCreateService(
    private val exampleRepository: ExampleRepository
) {

    fun start(exampleDo: ExampleDo): ResponseDto<Void> {
        exampleRepository.save(exampleDo)
        return ResponseDto(status = true)
    }
}
